crates/bevy_asset/src/asset_server.rs

```diff
@@ -361,11 +361,27 @@ impl AssetServer {
         asset_path.into()
     }
 
+    /// Loads all assets in the specified folder at `path` that satisfy the predicate `f`.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    /// use bevy_asset::AssetServer;
+    ///
+    /// fn load_png_files(asset_server: Res<AssetServer>) {
+    ///     let handles = asset_server.load_folder_if("my-assets/", |p| p.ends_with("png"));
```

Comment:
to check on extension, using [`extension`](https://doc.rust-lang.org/std/path/struct.Path.html#method.extension) is probably better